### PR TITLE
fix(kafka.js): wrap producer send function with shimmer

### LINKED ISSUE #284: kafka send: Maximum call stack size exceeded
version: epagon-node 1.63.1

When using `kafkajs#send` espagon produces an uncaught "Maximum call stack size exceeded" error that breaks our openwhisk action sending the event.

Downgrading to version 1.62.3 fixeds the issue which was probably introduced via #277 

https://github.com/epsagon/epsagon-node/compare/v1.62.3...v1.63.1

There seems to be circularity between:
```js
        const patchedSend = messages => kafkaSendMiddleware(messages, producer);
        ...
        response = producer.originalSend(messages);
```
in https://github.com/epsagon/epsagon-node/blob/master/src/events/kafkajs.js

Above two lines are featured in the stack:
```
RangeError: Maximum call stack size exceeded
    at Object.addException (/code/dist/action/index.js:43842:15)
    at Object.setException (/code/dist/action/index.js:43493:16)
    at kafkaSendMiddleware (/code/dist/action/index.js:57588:19)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
    at kafkaSendMiddleware (/code/dist/action/index.js:57585:29)
    at Object.patchedSend (/code/dist/action/index.js:57702:41)
```

/cc @ekremney @ranrib @idonava

### PR DESCRIPTION
**Pull request by idonava**
fixes: https://github.com/epsagon/epsagon-node/issues/284

### CODE REVIEW

**Review by adavidai — COMMENTED** [review]
Looks good 👍 
How did you test it?

**Review by adavidai — DISMISSED** [review]

File: src/events/kafkajs.js [hunk 1]
```diff
@@ -157,18 +156,19 @@ function kafkaConnectWrapper(originalConnectFunction) {
...
     return function internalKafkaProducerWrapper(options) {
-        // patching send request.
-        const producer = wrappedFunction.apply(this, [options]);
-        const patchedSend = messages => kafkaSendMiddleware(messages, producer);
-        producer.originalSend = producer.send;
-        producer.send = patchedSend.bind(producer);
-        return producer;
+        const producerResponse = producerFunction.apply(this, [options]);
+        try {
+            shimmer.wrap(producerResponse, 'send', () => wrapKafkaSendFunction(producerResponse.send));
```
Comment (ranrib) [inline src/events/kafkajs.js]: Can you add a protection so this shimmer will happen only once?

Comment (idonava) [reply src/events/kafkajs.js]: added

**Review by ranrib — APPROVED** [review]

**Review by adavidai — APPROVED** [review]

**Pull request merged**